### ISSUE TITLE
When @moto.mock_dynamodb2 is used then Dynamodb queries reaches AWS.

### ISSUE THREAD

**Issue by Grizmol** [spulec/moto#1818]
When @moto.mock_dynamodb2 is used then dynamodb queries reaches AWS.
When @moto.mock_dynamodb2_deprecated is used then dynamodb queries are mocked.

**Sample code to reproduce error**
```
import boto3, os, unittest, moto

@moto.mock_dynamodb2
class TestDynamoDb(unittest.TestCase):

	def test_dynamo_db(self):
		self.mock_table()

	@staticmethod
	def dynamo_db_resource():
		return boto3.resource('dynamodb', region_name='eu-west-2')

	@staticmethod
	def dynamo_db_table():
		return TestDynamoDb.dynamo_db_resource().Table("flights_table_test")

	def mock_table(self):
		TestDynamoDb.dynamo_db_resource().create_table(
			AttributeDefinitions = [
			    {
			        "AttributeName": "flight_identifier",
			        "AttributeType": "S"
			    },
			    {
			        "AttributeName": "from_airport_iata_code",
			        "AttributeType": "S"
			    },
			    {
			        "AttributeName": "price",
			        "AttributeType": "N"
			    },
			    {
			        "AttributeName": "start_date_timestamp",
			        "AttributeType": "S"
			    },
			    {
			        "AttributeName": "to_airport_iata_code",
			        "AttributeType": "S"
			    }
			],
			TableName = "flights_table_test",
			KeySchema = [
			    {
			        "AttributeName": "from_airport_iata_code",
			        "KeyType": "HASH"
			    },
			    {
			        "AttributeName": "flight_identifier",
			        "KeyType": "RANGE"
			    }
			],
			ProvisionedThroughput = {
			    "ReadCapacityUnits": 2,
			    "WriteCapacityUnits": 4
			},
			LocalSecondaryIndexes = [
			    {
			        "IndexName": "from_airport_iata_code-start_date_timestamp-index",
			        "KeySchema": [
			            {
			                "AttributeName": "from_airport_iata_code",
			                "KeyType": "HASH"
			            },
			            {
			                "AttributeName": "start_date_timestamp",
			                "KeyType": "RANGE"
			            }
			        ],
			        "Projection": {
			            "ProjectionType": "ALL"
			        },
			    },
			    {
			        "IndexName": "from_airport_iata_code-to_airport_iata_code-index",
			        "KeySchema": [
			            {
			                "AttributeName": "from_airport_iata_code",
			                "KeyType": "HASH"
			            },
			            {
			                "AttributeName": "to_airport_iata_code",
			                "KeyType": "RANGE"
			            }
			        ],
			        "Projection": {
			            "ProjectionType": "ALL"
			        },
			    },
			    {
			        "IndexName": "from_airport_iata_code-price-index",
			        "KeySchema": [
			            {
			                "AttributeName": "from_airport_iata_code",
			                "KeyType": "HASH"
			            },
			            {
			                "AttributeName": "price",
			                "KeyType": "RANGE"
			            }
			        ],
			        "Projection": {
			            "ProjectionType": "ALL"
			        },
			    }
			]
		)
```

**Expected result**
Query should be mocked.

**Actual error**
```
botocore.errorfactory.ResourceInUseException: An error occurred (ResourceInUseException) when calling the CreateTable operation: Table already exists: flights_table_test
```
**Full stack trace**

```
Traceback (most recent call last):
  File "/Users/piotrolechnowicz/Library/Python/3.6/lib/python/site-packages/moto/core/models.py", line 71, in wrapper
    result = func(*args, **kwargs)
  File "/Users/piotrolechnowicz/Desktop/motodeprecated_tests.py", line 7, in test_dynamo_db
    self.mock_table()
  File "/Users/piotrolechnowicz/Library/Python/3.6/lib/python/site-packages/moto/core/models.py", line 71, in wrapper
    result = func(*args, **kwargs)
  File "/Users/piotrolechnowicz/Desktop/motodeprecated_tests.py", line 102, in mock_table
    "ProjectionType": "ALL"
  File "/Users/piotrolechnowicz/Library/Python/3.6/lib/python/site-packages/boto3/resources/factory.py", line 520, in do_action
    response = action(self, *args, **kwargs)
  File "/Users/piotrolechnowicz/Library/Python/3.6/lib/python/site-packages/boto3/resources/action.py", line 83, in __call__
    response = getattr(parent.meta.client, operation_name)(**params)
  File "/Users/piotrolechnowicz/Library/Python/3.6/lib/python/site-packages/botocore/client.py", line 320, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/piotrolechnowicz/Library/Python/3.6/lib/python/site-packages/botocore/client.py", line 623, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.ResourceInUseException: An error occurred (ResourceInUseException) when calling the CreateTable operation: Table already exists: flights_table_test
```

**Library versions**
moto : 1.3.5
boto3 : 1.9.0

**Comment by Grizmol** [comment]
@joguSD thank you, 

For everyone who come here in future:
We have to wait for the new moto version to come. botocore >= 1.11.0 network code has changed. They no longer uses requests and instead directly uses urllib3. Moto 1.3.5 still tries to mock requests library.

Possible workarounds:
Pin version of boto3 to 1.7.84 or use @moto.mock_dynamodb2_deprecated decorator.

**Comment by spulec** [comment]
Closed in favor of #1793

**Comment by lp060** [comment]
Hi,

I am using moto.mock_dynamodb2_deprecated decorator and the following versions.

boto3==1.9.48
botocore==1.12.48
moto==1.3.7

I am trying to create a table but I get the following error:

An error occurred (ResourceInUseException) when calling the CreateTable operation: Resource in use
